Porting/updateAUTHORS.pl

```diff
@@ -1,661 +1,639 @@
+        my ($author_full)= Porting::updateAUTHORS->current_author_name_email("full");
+        my ($committer_full)= Porting::updateAUTHORS->current_committer_name_email("full");
 
-    while (defined(my $line= <$fh>)) {
-        chomp $line;
-        $line= decode_utf8($line);
-        my $commit_info= {};
-        @{$commit_info}{@field_names}= split /\0/, $line, 0 + @field_names;
+        push @{$opts{exclude_contrib}}, $author_full, $committer_full;
+    }
 
-        my $author_name_mm= _make_name_author_info($author_info, $commit_info,
-            "author_name_mm");
+    my $self= Porting::updateAUTHORS->new(%opts);
 
-        my $committer_name_mm=
-            _make_name_author_info($author_info, $commit_info,
-            "committer_name_mm");
+    my $changed= $self->read_and_update();
 
-        my $author_name_real= _make_name_simple($commit_info, "author");
+    if ($self->{validate}) {
+        is($self->unchanged_file($self->{$_}),1,
+            "Is '$self->{$_}' up to date?")
+            for qw(authors_file mailmap_file exclude_file);
 
-        my $committer_name_real= _make_name_simple($commit_info, "committer");
+        ok(!$self->{missing_author}{$_},
+            sprintf "%s is listed in AUTHORS", _clean_name($_))
+            for sort keys %{$self->{missing_author}||{}};
 
-        _check_name_mailmap(
-            $mailmap_info, $author_name_mm, $author_name_real,
-            $commit_info,  "author name"
-        );
-        _check_name_mailmap($mailmap_info, $committer_name_mm,
-            $committer_name_real, $commit_info, "committer name");
+        my $uncommitted_changes= $self->uncommitted_changes;
+        if ($uncommitted_changes) {
+            my ($author_name, $author_email)= $self->current_author_name_email();
+            my ($committer_name, $committer_email)= $self->current_committer_name_email();
 
-        $author_info->{"lines"}{$author_name_mm}++;
-        $author_info->{"lines"}{$committer_name_mm}++;
-    }
-    return $author_info;
-}
+            ok($author_name && $author_email,"git knows your author name and email.");
+            ok($committer_name && $committer_email,"git knows your author name and email.");
 
-sub read_authors {
-    my ($authors_file)= @_;
-    $authors_file ||= "AUTHORS";
-
-    my @authors_preamble;
-    open my $in_fh, "<", $authors_file
-        or die "Failed to open for read '$authors_file': $!";
-    while (defined(my $line= <$in_fh>)) {
-        chomp $line;
-        push @authors_preamble, $line;
-        if ($line =~ /^--/) {
-            last;
+            my $author_known= $self->known_contributor($author_name,$author_email);
+            my $committer_known= $self->known_contributor($committer_name,$committer_email);
+            if (is( $author_known && $committer_known, 1,
+                "Uncommitted changes are by a known contributor?"))
+            {
+                diag "Testing uncommtted changes! Remember to commit before you push!"
```

Comment:
I don't think it is helpful for this to `diag` at all.

Reply:
ok, ill remove it except under TEST_VERBOSE. FWIW, i think it makes sense to point out they aren't necesssarily testing what will be pushed up stream.  But its not a hill i want to die on.